test/test_helper.rb

```diff
@@ -2,46 +2,54 @@
 
 ENV["RAILS_ENV"] = "test"
 
+require 'warning'
+Warning.ignore(/.*/, %r{.*lib/shoulda/matchers/active_model/allow_value_matcher/attribute_setter.rb})
+Warning.ignore(/.*/, %r{.*lib/shoulda/context/context.rb})
+Warning.ignore(/.*/, %r{.*lib/mail/parsers.*})
```

Comment:
fyi - To address warnings such as the following that we don't have any control over that creates noise.

```
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/date_time_parser.rb:837: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/date_time_parser.rb:691: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32457: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32618: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32654: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32743: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32759: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32825: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32866: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32891: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:31987: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_disposition_parser.rb:795: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_disposition_parser.rb:819: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_disposition_parser.rb:586: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:761: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:592: warning: assigned but unused variable - disp_type_s
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:607: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_transfer_encoding_parser.rb:358: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_type_parser.rb:942: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_type_parser.rb:966: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/content_type_parser.rb:712: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/envelope_from_parser.rb:3241: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/message_ids_parser.rb:5112: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/message_ids_parser.rb:4848: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/mime_version_parser.rb:322: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/phrase_lists_parser.rb:702: warning: assigned but unused variable - testEof
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:7861: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8058: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8091: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8108: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8215: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8313: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8329: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8345: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8419: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8437: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8455: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8541: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8568: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8614: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8647: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8665: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8685: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8707: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8727: warning: statement not reached
/Users/dharam/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:7514: warning: assigned but unused variable - testEof
```